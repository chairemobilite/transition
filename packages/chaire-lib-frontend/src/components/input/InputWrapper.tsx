/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';

export type InputProps = {
    label: string;
    help?: string;
    smallInput?: boolean;
    twoColumns?: boolean;
};

export type HelpProps = {
    help: string;
};

const InputHelp: React.FunctionComponent<HelpProps> = (props: HelpProps) => {
    return <p className="_em _pale _small _help">{props.help}</p>;
};

export const InputWrapper: React.FunctionComponent<React.PropsWithChildren<InputProps>> = ({
    label,
    help,
    smallInput = false,
    children,
    twoColumns = true
}: React.PropsWithChildren<InputProps>) => {
    const classes = `apptr__form-input-container${twoColumns ? ' _two-columns' : ''}${
        smallInput ? ' _small-inputs' : ''
    }`;
    return (
        <React.Fragment>
            <div className={classes}>
                <label>{label}</label>
                {children}
                {help && (
                    <React.Fragment>
                        <div className={'_flex-break'}></div>
                        <InputHelp help={help}></InputHelp>
                    </React.Fragment>
                )}
            </div>
        </React.Fragment>
    );
};

export default InputWrapper;
