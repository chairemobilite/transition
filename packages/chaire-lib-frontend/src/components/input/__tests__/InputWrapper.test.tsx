/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import * as React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import InputWrapper from '../InputWrapper';

const testLabel = 'Test';

test('Basic wrapper', () => {
    const { container } = render(<InputWrapper
        label = {testLabel}>
        <input id="test"/>
    </InputWrapper>);
    expect(container).toMatchSnapshot();
});

test('Wrapper with help', () => {
    const { container } = render(<InputWrapper
        label = {testLabel}
        help = "My help text">
        <input id="test"/>
    </InputWrapper>);
    expect(container).toMatchSnapshot();
});

test('Wrapper with small box and multiple children', () => {
    const { container } = render(<InputWrapper
        label = {testLabel}
        smallInput = {true}>
        <input id="test"/>
        <input id="test2"/>
    </InputWrapper>);
    expect(container).toMatchSnapshot();
});

test('Wrapper in one column', () => {
    const { container } = render(<InputWrapper
        label = {testLabel}
        twoColumns = {false}>
        <input id="test"/>
    </InputWrapper>);
    expect(container).toMatchSnapshot();
});

test('Wrapper in one column and small', () => {
    const { container } = render(<InputWrapper
        label = {testLabel}
        twoColumns = {false}
        smallInput = {true}>
        <input id="test"/>
    </InputWrapper>);
    expect(container).toMatchSnapshot();
});
