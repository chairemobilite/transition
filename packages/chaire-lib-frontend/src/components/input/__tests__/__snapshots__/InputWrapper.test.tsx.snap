// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic wrapper 1`] = `
<div>
  <div
    class="apptr__form-input-container _two-columns"
  >
    <label>
      Test
    </label>
    <input
      id="test"
    />
  </div>
</div>
`;

exports[`Wrapper in one column 1`] = `
<div>
  <div
    class="apptr__form-input-container"
  >
    <label>
      Test
    </label>
    <input
      id="test"
    />
  </div>
</div>
`;

exports[`Wrapper in one column and small 1`] = `
<div>
  <div
    class="apptr__form-input-container _small-inputs"
  >
    <label>
      Test
    </label>
    <input
      id="test"
    />
  </div>
</div>
`;

exports[`Wrapper with help 1`] = `
<div>
  <div
    class="apptr__form-input-container _two-columns"
  >
    <label>
      Test
    </label>
    <input
      id="test"
    />
    <div
      class="_flex-break"
    />
    <p
      class="_em _pale _small _help"
    >
      My help text
    </p>
  </div>
</div>
`;

exports[`Wrapper with small box and multiple children 1`] = `
<div>
  <div
    class="apptr__form-input-container _two-columns _small-inputs"
  >
    <label>
      Test
    </label>
    <input
      id="test"
    />
    <input
      id="test2"
    />
  </div>
</div>
`;
